core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -178,10 +178,9 @@ public int formatVersion() {
   }
 
   private int formatVersion(Table table) {
-    if (table instanceof HasTableOperations) {
-      HasTableOperations ops = (HasTableOperations) table;
-      return ops.operations().current().formatVersion();
-    } else {
+    try {
+      return TableUtil.formatVersion(table);
+    } catch (IllegalArgumentException e) {
```

Comment:
note that the try-catch is mainly so that we don't fail during the instantiation of `SerializableTable`: https://github.com/apache/iceberg/pull/12462/files#diff-46dafb425240806166ccc8f27a6c301781c5082bf1a187291cc92c2a8f17a588R90